misc/server/Dockerfile

```diff
@@ -1,19 +1,20 @@
-FROM rust:1.72-bullseye as builder
-WORKDIR /usr/src/rust-libp2p-server
+# syntax=docker/dockerfile:1.5-labs
+FROM rust:1.67.0 as chef
+RUN wget -q -O- https://github.com/LukeMathWalker/cargo-chef/releases/download/v0.1.62/cargo-chef-x86_64-unknown-linux-gnu.tar.gz | tar -zx -C /usr/local/bin
+RUN cargo install --locked --root /usr/local libp2p-lookup --version 0.6.4
+WORKDIR /app
 
-# Run with access to the target cache to speed up builds
-WORKDIR /workspace
+FROM chef AS planner
+COPY . .
```

Comment:
Wouldn't this cause a re-build of the layer each time anything in the working directory changes? I think I am missing something.

For others, this is the way to do it according to https://github.com/LukeMathWalker/cargo-chef#how-to-use.

Reply:
Yes, this layer is being rebuilt but the output of this layer is always the same, i.e. as long as your dependencies don't change, you get the same `recipe.json`. Note that the next layer starts again from `chef` and not from `planner`. We just copy over `recipe.json` from `planner` which will not change unless our dependencies change and thus the next layer is always cached.